lib/src/http.dart

```diff
@@ -210,8 +212,36 @@ class _PubHttpClient extends http.BaseClient {
 /// The [_PubHttpClient] wrapped by [httpClient].
 final _pubClient = new _PubHttpClient();
 
+/// A set of all hostnames for which we've printed a message indicating that
+/// we're waiting for them to come back up.
+final _retryMessages = new Set<String>();
+
 /// The HTTP client to use for all HTTP requests.
-final httpClient = new ThrottleClient(16, _pubClient);
+final httpClient = new ThrottleClient(
+    16,
+    new RetryClient(_pubClient,
+        retries: 5,
+        when: (response) => const [502, 503, 504].contains(response.statusCode),
+        delay: (retryCount) {
+          if (retryCount < 3) {
+            // Retry quickly a couple times in case of a short transient error.
+            return new Duration(milliseconds: 500) * math.pow(1.5, retryCount);
```

Comment:
It is a good idea to also add in an additional random delay - you would keep exactly this logic but just add between maybe 0 and 500 additional milliseconds. This helps to ensure that when a lot of requests are sent at once they don't all keep rescheduling on the same intervals.

Reply:
Done.